### PR TITLE
Propagate the selected facet/agg value up to the highest parent component

### PR DESCRIPTION
**Pull request by bdolly**
When a user interacts with a facet/aggregation value propagate that call and it's value up to the highest parent component to allow for hooks into these events

### CODE REVIEW

**Comment by alex-wilmer** [comment]
looks like your editor is ignoring the prettier config options. do you have eslint installed in your editor?

```
  'prettier/prettier': [1, { trailingComma: 'all', singleQuote: true }],
```

File: modules/components/src/Arranger/Aggregations.js [hunk 1]
```diff
@@ -1,22 +1,23 @@
-import React from 'react';
+import React from "react";
```
Comment (alex-wilmer) [inline modules/components/src/Arranger/Aggregations.js]: prettier here too?

**Comment by alex-wilmer** [comment]
I don't see a problem with this. @jgnieuwhof you wrote most of these components, what do you think?

**Comment by hlminh2000** [comment]
my only thought is `field` should all be available from the wrapping components in `modules/components/src/Aggs/aggComponentsMap.js` and fed down, so it doesn't need to be propagated all the way up from the individual agg components

File: modules/components/src/Aggs/RangeAgg.js [hunk 2]
```diff
@@ -1,18 +1,18 @@
-import React, { Component } from 'react';
-import InputRange from 'react-input-range';
-import convert from 'convert-units';
-import _ from 'lodash';
+import React, { Component } from "react";
```
Comment (alex-wilmer) [inline modules/components/src/Aggs/RangeAgg.js]: oh more prettier stuff

**Comment by alex-wilmer** [comment]
I think this change just adds that information on events. It's true could probably grab the same info inside the function body of the event where it's called, but this way it's just there?

**Comment by hlminh2000** [comment]
yeah it doesn't make any difference for the data collection, would just be less code :P 
I'm ok with having it coming from the event too if that's too much hassle to change now :)

**Comment by bdolly** [comment]
I like having it in the events because it provides "hooks" for the app that it's nested inside of since Arranger is typically used in a sub-app context. 

The main use case right now is hooking in analytics events to the various user interactions

**Comment by hlminh2000** [comment]
ok that's cool for me then :)

**Pull request merged**